python_modules/dagster/dagster/_daemon/run_coordinator/queued_run_coordinator_daemon.py

```diff
@@ -270,48 +353,131 @@ def get_priority(run):
         # sorted is stable, so fifo is maintained
         return sorted(runs, key=get_priority, reverse=True)
 
-    def _dequeue_run_guarded(
+    def _is_location_pausing_dequeues(self, location_name, now):
+        with self._location_timeouts_lock:
+            return (
+                location_name in self._location_timeouts
+                and self._location_timeouts[location_name] > now
+            )
+
+    def _dequeue_run(
         self,
-        workspace_process_context: IWorkspaceProcessContext,
+        instance: DagsterInstance,
+        workspace: IWorkspace,
         run: DagsterRun,
-    ) -> Optional[SerializableErrorInfo]:
-        instance = workspace_process_context.instance
-        error_info = None
+        run_queue_config: RunQueueConfig,
+        fixed_iteration_time: Optional[float],
+    ) -> bool:
 
-        try:
-            self._dequeue_run(workspace_process_context, run)
-        except Exception:
-            error_info = serializable_error_info_from_exc_info(sys.exc_info())
+        # double check that the run is still queued before dequeing
+        run = check.not_none(instance.get_run_by_id(run.run_id))
+
+        now = fixed_iteration_time or time.time()
 
-            message = (
-                f"Caught an error for run {run.run_id} while removing it from the queue."
-                " Marking the run as failed and dropping it from the queue"
+        if run.status != DagsterRunStatus.QUEUED:
+            self._logger.info(
+                "Run %s is now %s instead of QUEUED, skipping",
+                run.run_id,
+                run.status,
             )
-            message_with_full_error = f"{message}: {error_info.to_string()}"
+            return False
 
-            self._logger.error(message_with_full_error)
-            instance.report_run_failed(run, message_with_full_error)
+        location_name = (
+            run.external_pipeline_origin.location_name if run.external_pipeline_origin else None
+        )
```

Comment:
in what conditions is this None?

Reply:
Pretty much impossible now, maybe runs enqueued back on like 0.9.0

Reply:
or maybe runs programatically submitted in python somehow

Reply:
think its worth leave a comment here